packages/protocol/lib/backward/ast-code.ts

```diff
@@ -0,0 +1,392 @@
+// tslint:disable: max-classes-per-file
+import { makeZContract } from '@celo/protocol/lib/backward/internal';
+import {
+  BuildArtifacts,
+  Contract as ZContract
+} from '@openzeppelin/upgrades';
+import ContractAST from '@openzeppelin/upgrades/lib/utils/ContractAST';
+
+const VISIBILITY_PUBLIC = 'public'
+const VISIBILITY_EXTERNAL = 'external'
+const CONTRACT_TYPE_CONTRACT = 'contract'
+const STORAGE_DEFAULT = 'default'
+
+const OUT_VOID_PARAMETER_STRING = 'void'
+
+export class ASTCodeCompatibilityReport {
+  changes: Change[];
+  constructor(changes: Change[]) {
+    this.changes = changes;
+  }
+  public push(...changes: Change[]) {
+    this.changes.push(...changes)
+  }
+  public include(other: ASTCodeCompatibilityReport) {
+    this.push(...other.changes)
```

Comment:
```suggestion
    this.changes.push(...other.changes)
```

Reply:
I think this is a left-over from my scala/java days. The reasoning usually on this type of code is that you leave space for someone to actually extend & modify the 'push' function, without needing to modify the 'pushAll' (include) function.